R/insertPackage.R

```diff
@@ -159,75 +159,68 @@ identifyPackageType <- function(file) {
     return(ret)
 }
 
+##' This function returns the compile-time information added
+##' to the \code{DESCRIPTION} file in the package.
+##'
+##' @title Get information from a binary package 
+##' @param file the fully qualified path of the package
+##' @return A named vector with several components
+##' @author Dirk Eddelbuettel
+getPackageInfo <- function(file) {
+    if (!file.exists(file)) stop("File ", file, " not found!", call.=FALSE)
+    
+    td <- tempdir()
+    if (grepl(".zip$", file)) {
+        unzip(file, exdir=td)
+    } else if (grepl(".tgz$", file)) {
+        untar(file, exdir=td)
+    } else {
+        ##stop("Not sure we can handle ", file, call.=FALSE)
+        fields <- c("Source"=TRUE, "Rmajor"=NA, "Mavericks"=FALSE)
+        return(fields)
+    }
+
+    pkgname <- gsub("^([a-zA-Z0-9]*)_.*", "\\1", basename(file))
+    path <- file.path(td, pkgname, "DESCRIPTION")
+    builtstring <- read.dcf(path, 'Built')
+    unlink(file.path(td, pkgname), recursive=TRUE)
+
+    fields <- strsplit(builtstring, "; ")[[1]]
+    names(fields) <- c("Rversion", "OSflavour", "Date", "OS")
+
+    rmajor <- gsub("^R (\\d\\.\\d)\\.\\d.*", "\\1", fields["Rversion"])
+    isDarwin13 <- ifelse(fields["OSflavour"] == "x86_64-apple-darwin13.4.0", "yes", "no")
+    fields <- c(fields, "Rmajor"=unname(rmajor), "Mavericks"=unname(isDarwin13))
```

Comment:
I am not sure if this was an attempt at a joke or not, and marginal usefulness of this discussion has declined considerably a long time ago.   

The warning would be about a package built (by whomever/wherever) on a system older than the current OS X.

I value your input here, but I fear we are wasting both your time and mine.  If you have *concreate* proposals that can be *implemented* then please consider a pull request.

Otherwise, I suggest we end this now and here.